lib/cog.ex

```diff
@@ -61,4 +62,45 @@ defmodule Cog do
   def adapter_module(bad_adapter) do
     {:error, "The adapter is set to '#{bad_adapter}', but I don't know what that is. Try 'slack' or 'hipchat' instead."}
   end
+
+  defp sanity_check() do
+    {smp_status, smp_message} = verify_smp()
+    {ds_status, ds_message} = verify_dirty_schedulers()
+    if smp_status == :ok do
+      Logger.info(smp_message)
+    else
+      Logger.error(smp_message)
+    end
+    if ds_status == :ok do
+      Logger.info(ds_message)
+    else
+      Logger.error(ds_message)
+    end
+    unless smp_status == :ok and ds_status == :ok do
+      Logger.error("Application start aborted.")
+      Logger.flush()
+      :init.stop()
+    end
+  end
+
+  defp verify_smp() do
+    if :erlang.system_info(:schedulers_online) < 2 do
+      {:error, "SMP support disabled. Add '-smp enable' to $ERL_FLAGS and restart Cog."}
```

Comment:
Since we're not using `ERL_FLAGS` to enable this when run with `iex` we could add that info in here as well.